k9mail/src/main/java/com/fsck/k9/Preferences.java

```diff
@@ -164,4 +164,17 @@ public void setDefaultAccount(Account account) {
     public SharedPreferences getPreferences() {
         return mStorage;
     }
+
+    public static <T extends Enum<T>> T getEnumStringPref(SharedPreferences prefs, String key, T defaultEnum) {
+        String stringPref = prefs.getString(key, defaultEnum.name());
+        try {
+            return Enum.valueOf(defaultEnum.getDeclaringClass(), stringPref);
+        } catch (Exception ex) {
```

Comment:
should not happen since a default gets passed in, unless the preference got explicitly set to null.